core/hash/mini.odin

```diff
@@ -31,10 +31,65 @@ ginger8 :: proc "contextless" (data: []byte) -> u8 {
 	}
 	return h
 }
+
 ginger16 :: proc "contextless" (data: []byte) -> u16 {
 	h := ginger_hash16(0)
 	for b in data {
 		h ~= ginger_hash16(u16(b))
 	}
 	return h
 }
+
+
+@(private)
+sxm_hash_uint_generic :: #force_inline proc "contextless" (x: $T) -> T {
+	bits :: size_of(x) << 3
+	shift :: bits >> 1
+	mul :: 0x4ff55ba64bb740e135db2be3690a61d3 % (1 << bits)
+	o := T(x)
+	o = (o ~ o >> shift) * mul
+	o = (o ~ o >> shift) * mul
+	o = (o ~ o >> shift) * mul
+	o = (o ~ o >> shift) * mul
+	return o
+}
+
+sxm_hash8 :: proc "contextless" (x: u8) -> u8 {
+	return sxm_hash_uint_generic(x)
+}
+
+sxm_hash16 :: proc "contextless" (x: u16) -> u16 {
+	return sxm_hash_uint_generic(x)
+}
+
+sxm_hash32 :: proc "contextless" (x: u32) -> u32 {
+	return sxm_hash_uint_generic(x)
+}
+
+sxm_hash64 :: proc "contextless" (x: u64) -> u64 {
+	return sxm_hash_uint_generic(x)
+}
+
+sxm_hash128 :: proc "contextless" (x: u128) -> u128 {
+	return sxm_hash_uint_generic(x)
+}
+
+sxm_hash_uint :: proc {
```

Comment:
I'd remove the `_uint` suffix as someone might think it is actually for a `uint` type.

Reply:
fixed both formatting and the suffix